### PR TITLE
I2C Integration

### PR DESCRIPTION
**Pull request by sapphire-arches**
PR from the I2C branch so I can leave comments

### CODE REVIEW

File: board_common/common/i2c.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#include "i2c.h"
+/******************************************************************************\
+ *  Public interface implementations                                          *
+\******************************************************************************/
+
+i2c_error_t i2c_write_bytes(i2_t * channel, const uint8_t * bytes, size_t n) {
+    i2c_error_t err;
+    for (size_t i = 0; i < n; i++) {
+        err = uart_write_byte(channel, bytes[i]);
```
Comment (sapphire-arches) [inline board_common/common/i2c.c:9]: should be `i2c_write_byte`

File: board_common/common/i2c.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        if (err != I2C_NO_ERROR)
+            return err;
+    }
+    return err;
+}
+
+i2c_error_t i2c_read_bytes(i2_t * channel, const uint8_t * bytes, size_t n) {
+    i2c_error_t err;
+    for (size_t i = 0; i < n; i++) {
+        err = uart_read_byte(channel, bytes[i]);
```
Comment (sapphire-arches) [inline board_common/common/i2c.c:19]: should be `i2c_read_byte`

File: board_common/native/i2c_eusci_native.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    EUSCI_B1,
+#endif
+#ifdef EUSCI_B2_BASE
+    EUSCI_B2,
+#endif
+#ifdef EUSCI_B3_BASE
+    EUSCI_B3,
+#endif
+    EUSCI_count
+} eusci_t;
```
Comment (sapphire-arches) [inline board_common/native/i2c_eusci_native.h:51]: I think the time to resolve that TODO is probably now, either in this branch or in @branw's SPI branch

**Review by sapphire-arches — COMMENTED** [review]
I left a few comments but this isn't really complete enough yet to do a full review on.

**Comment by sapphire-arches** [comment]
Things that need doing:
  - Clean up all those TODOs
  - Add tests!

File: board_common/native/i2c_eusci_native.h (hunk 3 above)
Comment (branw) [reply board_common/native/i2c_eusci_native.h:51]: Fixed in 40d74ff2bbfdeb6b6cd29f630048d17bc47b1fa0 for #22

**Comment by branw** [comment]
Moved into #26

**Pull request closed without merging**